optuna/study.py

```diff
@@ -201,6 +202,8 @@ def __init__(
 
         self.sampler = sampler or samplers.TPESampler()
         self.pruner = pruner or pruners.MedianPruner()
+        if isinstance(self.pruner, pruners.HyperbandPruner):
+            self.sampler = pruners.hyperband._HyperbandSampler(self.sampler, self.pruner)
```

Comment:
How about using custom getters and setters for `self.sampler` (and internally store `self._sampler = _HyperbandSampler`)? `Study.sampler` is a part of the public interface after all.

Reply:
I implemented the custom getter and setter of `_HyperbandSampler` in `hyperband.py`. I think it addresses the suggestion as pointed out.

Reply:
Hm, I'm not so sure about that. `study.sampler` here return an instance of `_HyperbandSampler`. I was imagining having `study._sampler` being the `_HyperbandSampler` while `study.sampler` being a custom, in this case, accessor that return `study._sampler.sampler` which is the original sampler object given by the user. Similarly for the setter.

Reply:
Note that we must also update `Trial._suggest` and `Trial._init_relative_params`.